pkg/install/app/vaultconfigdeps.go

```diff
@@ -0,0 +1,239 @@
+		vd.ConfigJob,
+	}
+
+	for _, laf := range lafs {
+		for label, value := range vd.Labels {
+			lifecycle.Label(ctx, laf, label, value)
+		}
+	}
+
+	ConfigureVaultConfigJob(
+		vd.Core.Key,
+		vd.Core.Object.Spec.LogService,
+		vd.Core.Object.Spec.MetadataAPI,
+		vd.Core.Object.Spec.Operator,
+		vd.Core.Object.Spec.Vault,
+		vd.ConfigJob, vd.JWTSigningKeySecret)
+
+	return nil
+}
+
+func NewVaultConfigDeps(c *obj.Core, jwt *corev1obj.Secret) *VaultConfigDeps {
+	vd := &VaultConfigDeps{
+		Core:                c,
+		JWTSigningKeySecret: jwt,
+		Labels: map[string]string{
+			model.RelayInstallerNameLabel: c.Key.Name,
+			model.RelayAppManagedByLabel:  "relay-install-operator",
+		},
+	}
+
+	return vd
+}
+
+func ConfigureVaultConfigJob(
+	coreKey types.NamespacedName,
+	logServiceConfig v1alpha1.LogServiceConfig,
+	metadataAPIConfig *v1alpha1.MetadataAPIConfig,
+	operatorConfig *v1alpha1.OperatorConfig,
+	vaultConfig *v1alpha1.VaultConfig,
+	job *batchv1obj.Job, jwt *corev1obj.Secret) {
+
+	// FIXME Determine best approach to handling slight variations in configuration data
+	authPath := strings.Split(metadataAPIConfig.VaultAuthPath, "/")
+
+	env := []corev1.EnvVar{
+		{Name: logServicePathEnvVar, Value: vaultConfig.LogServicePath},
+		{Name: logServiceVaultAgentRoleEnvVar, Value: logServiceConfig.VaultAgentRole},
+		{Name: metadataAPIVaultAgentRoleEnvVar, Value: metadataAPIConfig.VaultAgentRole},
+		{Name: operatorVaultAgentRoleEnvVar, Value: operatorConfig.VaultAgentRole},
+		{Name: tenantPathEnvVar, Value: vaultConfig.TenantPath},
+		{Name: transitKeyEnvVar, Value: vaultConfig.TransitKey},
+		{Name: transitPathEnvVar, Value: vaultConfig.TransitPath},
+		{Name: vaultAddrEnvVar, Value: vaultAddr},
+		{Name: vaultJWTAuthPathEnvVar, Value: metadataAPIConfig.VaultAuthPath},
+		{Name: vaultJWTMountEnvVar, Value: authPath[len(authPath)-1]},
+		{Name: vaultNameEnvVar, Value: vaultIdentifier},
+		{Name: vaultNamespaceEnvVar, Value: coreKey.Namespace},
+		{Name: vaultServiceAccountEnvVar, Value: vaultConfig.AuthDelegatorServiceAccountName},
+	}
+
+	if jwt != nil {
+		env = append(env,
+			corev1.EnvVar{
+				Name: vaultJWTPublicKeyEnvVar,
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						Key: defaultPublicJWTSigningKeyName,
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: jwt.Key.Name,
+						},
+					},
+				},
+			},
+		)
+	}
+
+	if vaultConfig.Auth != nil {
+		if token, ok := VaultAuthDataEnvVar(vaultTokenEnvVarName, vaultConfig.Auth.Token); ok {
+			env = append(env, token)
+		}
+
+		if unsealKey, ok := VaultAuthDataEnvVar(vaultUnsealKeyEnvVarName, vaultConfig.Auth.UnsealKey); ok {
+			env = append(env, unsealKey)
+		}
+	}
+
+	job.Object.Spec.Template.Spec.Containers = []corev1.Container{
+		{
+			Name:            vaultInitIdentifier,
+			Image:           vaultConfig.Image,
+			ImagePullPolicy: vaultConfig.ImagePullPolicy,
+			Env:             env,
```

Comment:
I think -- but I'm not 100% sure -- that once a job has been kicked off, its spec essentially becomes immutable, so if you try to update the CRD you might find that this will cause an infinite loop from being unable to persist the job changes. Goes back to our initialization process reconciler conversation earlier. For a quick fix, you could perhaps generate the whole spec early, serialize + hash it, and append it to the job name. You'd probably end up with some old jobs left lingering but that could be cleaned up later (might want to slap a good label on the metadata though to easily identify them for the future).